include/boost_plugin_loader/plugin_loader.h

```diff
@@ -146,7 +153,13 @@ class PluginLoader
    */
   inline bool empty() const;
 
+  /** @brief Clear internal state including cache objects */
+  inline void clear();
+
 protected:
+  mutable std::mutex libraries_mutex_;
+  mutable std::unordered_map<std::string, boost::dll::shared_library> libraries_by_path_;
```

Comment:
Small rename and docs for clarity

```suggestion
  /**
  * @brief Internal cache of loaded plugin libraries, stored by the path from which the library was loaded
  */
  mutable std::unordered_map<std::string, boost::dll::shared_library> libraries_;
```